src/fsharp/Optimizer.fs

```diff
@@ -2398,89 +2397,65 @@ and TryDevirtualizeApplication cenv env (f,tyargs,args,m) =
     // Optimize/analyze calls to LanguagePrimitives.HashCompare.GenericComparisonWithComparerIntrinsic for tuple types
     | Expr.Val(v,_,_),[ty],_ when  valRefEq cenv.g v cenv.g.generic_comparison_inner_vref && isRefTupleTy cenv.g ty ->
         let tyargs = destRefTupleTy cenv.g ty 
-        let vref = 
-            match tyargs.Length with 
-            | 2 -> Some cenv.g.generic_compare_withc_tuple2_vref 
-            | 3 -> Some cenv.g.generic_compare_withc_tuple3_vref 
-            | 4 -> Some cenv.g.generic_compare_withc_tuple4_vref 
-            | 5 -> Some cenv.g.generic_compare_withc_tuple5_vref 
-            | _ -> None
-        match vref with 
-        | Some vref -> Some (DevirtualizeApplication cenv env vref ty tyargs (mkCallGetGenericComparer cenv.g m :: args) m)            
-        | None -> None
+        match tyargs.Length with 
+        | 2 -> Some (DevirtualizeApplication cenv env cenv.g.generic_compare_withc_tuple2_vref ty tyargs (mkCallGetGenericComparer cenv.g m :: args) m)
+        | 3 -> Some (DevirtualizeApplication cenv env cenv.g.generic_compare_withc_tuple3_vref ty tyargs (mkCallGetGenericComparer cenv.g m :: args) m)
+        | 4 -> Some (DevirtualizeApplication cenv env cenv.g.generic_compare_withc_tuple4_vref ty tyargs (mkCallGetGenericComparer cenv.g m :: args) m)
+        | 5 -> Some (DevirtualizeApplication cenv env cenv.g.generic_compare_withc_tuple5_vref ty tyargs (mkCallGetGenericComparer cenv.g m :: args) m)
+        | _ -> None
```

Comment:
This diff duplicates code - it doesn't seem right unless we know for sure it gives a measurable performance win.